src/tests/sphdirectpcqueue_ttt.c

```diff
@@ -805,11 +820,18 @@ test_pcq_reset ()
 }
 
 int
-main ()
+main (int argc, char *argv[])
 {
   int rc, rc3;
   double clock, nano, rate;
   sphtimer_t tempt, startt, endt, freqt;
+  int i;
+
+  for (i = 1; i < argc; i++) {
+    cpu_list[i-1] = strtol(argv[i],0,0);
+    if (cpu_list[i-1] > cpu_max) cpu_max = cpu_list[i-1];
```

Comment:
No. The purpose of this code is to compute how big the CPU_SET needs to be.  cpu_max is later passed into CPU_ALLOC(). The only reason to compute cpu_max is to limit the size of the CPU_SET as allocated by CPU_ALLOC().  Also, given the gaps that appear in the CPU numbering on ppc64*, using _SC_NPROCESSORS_ONLN may be much less than the maximum CPU number, especially in SMT=1 mode.  If any absolute maximum were to be used, it should be _SC_NPROCESSORS_CONF.  Admittedly, we're talking about very little memory, either way.  I'm content to let the user specify whatever CPU numbers they wish. If you feel strongly that an absolute maximum should be added, I can do that.